lib/backend/k8s/conversion/conversion.go

```diff
@@ -430,7 +433,14 @@ func (c converter) k8sRuleToCalico(rPeers []networkingv1.NetworkPolicyPeer, rPor
 		}
 
 		pStr := protocol.String()
-		protocolPorts[pStr] = append(protocolPorts[pStr], calicoPorts...)
+		// treat nil as 'all ports'
+		if calicoPorts == nil {
```

Comment:
Comment above on line 426 is now out of date; `calicoPorts` can be nil, now.

Reply:
👍